src/utils/api.js

```diff
@@ -0,0 +1,25 @@
+import { baseUrl } from "./constants";
+import { processServerResopnse } from "./utils";
+
+const storeRecipe = (id) => {
+  return fetch(`${baseUrl}/users/recipes/${id}`, {
+    method: "PUT",
+    credentials: "include",
+  }).then(processServerResopnse);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(processServerResopnse)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.